File: string-paradigm-no-wrapper/src/main/java/com/jesperancinha/string/paradigm/nowrapper/performance/NoWrapperDependencyTableModel.java
package com.jesperancinha.string.paradigm.nowrapper.performance;

import java.util.ArrayList;
import java.util.List;

import javax.swing.table.AbstractTableModel;

import com.steelzack.string.paradigm.api.Dependency;
import com.steelzack.string.paradigm.api.SegmentTableModelIntReturn;


public class NoWrapperDependencyTableModel extends AbstractTableModel implements SegmentTableModelIntReturn{

    /**
	 * 
	 */
	private static final long serialVersionUID = 707044827159393606L;
	private List<Dependency> dependencies = new ArrayList<Dependency>();
    private int columnCount;
    
    @Override
    public int addCombination(Dependency dependency) {
        this.dependencies.add(dependency);
        
        int codeCount = dependency.getDependencyCount();
        if(codeCount != this.columnCount) {
            this.columnCount = Math.max(columnCount, codeCount);
        }
        
        int row = dependencies.size() - 1;
        return row;
    }

    @Override
    public int getRowCount() {
        return dependencies.size();
    }

    @Override
    public int getColumnCount() {
        return columnCount;
    }

    @Override
    public Object getValueAt(int rowIndex, int columnIndex) {
        Dependency dependency = dependencies.get(rowIndex);
        return dependency.getSubDependencyElement(columnIndex);
    }

    public void clear() {
        int size = this.dependencies.size();
        if(size > 0) {
            this.dependencies.clear();
            fireTableRowsDeleted(0, size - 1);
        }
    }
}

File: string-paradigm-no-wrapper/src/main/kotlin/org/jesperancinha/string/paradigm/nowrapper/performance/NoWrapperDependencyTableModel.kt
package org.jesperancinha.string.paradigm.nowrapper.performance

import org.jesperancinha.string.paradigm.api.Dependency
import org.jesperancinha.string.paradigm.api.SegmentTableModelIntReturn
import javax.swing.table.AbstractTableModel

class NoWrapperDependencyTableModel : AbstractTableModel(), SegmentTableModelIntReturn {
    private val dependencies: MutableList<Dependency> = ArrayList()
    private var columnCount = 0
    override fun addCombination(dependency: Dependency): Int {
        dependencies.add(dependency)
        val codeCount: Int = dependency.dependencyCount()
        if (codeCount != columnCount) {
            columnCount = columnCount.coerceAtLeast(codeCount)
        }
        return dependencies.size - 1
    }

    override fun getRowCount(): Int {
        return dependencies.size
    }

    override fun getColumnCount(): Int {
        return columnCount
    }

    override fun getValueAt(rowIndex: Int, columnIndex: Int): String? {
        val dependency: Dependency = dependencies[rowIndex]
        return dependency.getSubDependencyElement(columnIndex)
    }

    override fun clear() {
        val size = dependencies.size
        if (size > 0) {
            dependencies.clear()
            fireTableRowsDeleted(0, size - 1)
        }
    }

    companion object {
        /**
         *
         */
        private const val serialVersionUID = 707044827159393606L
    }
}